### PR TITLE
OSSM-89 - Add support for ES index cleaner configuration

### PR DESCRIPTION
**Pull request by objectiser**
Enables the `esIndexCleaner` node, with the [options documented here](https://www.jaegertracing.io/docs/1.13/operator/#elasticsearch-index-cleaner-job), to be added to the service mesh CR.

Signed-off-by: Gary Brown <gary@brownuk.com>

### CODE REVIEW

**Comment by objectiser** [comment]
cc @pavolloffay

File: helm/istio/charts/tracing/templates/jaeger-production-elasticsearch.yaml [hunk 1]
```diff
@@ -87,6 +87,11 @@ spec:
           memory: "16Gi"
 {{- end }}
 
+{{- if .Values.jaeger.esIndexCleaner }}
```
Comment (pavolloffay) [inline helm/istio/charts/tracing/templates/jaeger-production-elasticsearch.yaml:75]: The ES cleaner spec https://godoc.org/github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1#JaegerEsIndexCleanerSpec

A side effect approach is that some configuration can be provided in smcp CR and some in Jaeger CR. For example here users specify scheduled time but retention day can be specified in Jaeger CR.

Comment (objectiser) [reply helm/istio/charts/tracing/templates/jaeger-production-elasticsearch.yaml:75]: Not sure I follow - this template is used to generate the Jaeger CR, so it shouldn't be possible for the retention day to be specified unless it is defined in the smcp CR?

Comment (pavolloffay) [reply helm/istio/charts/tracing/templates/jaeger-production-elasticsearch.yaml:75]: The template is used to generate Jaeger CR, however the fields not specified in smcp CR can be directly edited in Jaeger CR and maistra operator would not change them bc it does not know about them. The smcp CR does not define what fields cleaner CR contains, it is a free form anything can be added there.

At least I remember this behavior from previous work on this CR. 

Also maybe condition `{{- if .Values.jaeger.esIndexCleaner }}` is redundant and cleaner could be passed there directly.

Comment (pavolloffay) [reply helm/istio/charts/tracing/templates/jaeger-production-elasticsearch.yaml:75]: I am not saying it is an issue, just behavior to think about.

**Review by pavolloffay — APPROVED** [review]

**Comment by objectiser** [comment]
@knrc @rcernich Could this be merged?

**Review by rcernich — APPROVED** [review]

**Review by dgn — APPROVED** [review]

**Comment by objectiser** [comment]
@rcernich @dgn Can this be merged now?

**Pull request merged**